### ISSUE TITLE
Screen redraw issue in main index

### ISSUE THREAD

**Issue by dcpurton** [neomutt/neomutt#3304]
This is one issue I'm noticing with the new event driven screen redraws.

I use a tiling WM and I see this as I add and remove terminal windows, which causes the NeoMutt terminal to resize.

The status bar does not always resize correctly.

For testing I'm using this `rc` file:

```ini
# vim: syn=neomuttrc
# filename: test.rc
# run from build directory:
#   ./neomutt -n -F test.rc
unset record
unset signature
set folder = "./"
set from = "john.doe@example.com"
set realname = "John Doe"
set spoolfile = "/dev/null"
```

Here's a series of screenshots highlighting the redraw issue. It occurs mostly, but not quite always. I can't seem to track down an exact pattern when it does work yet.

1. Full screen terminal (initial state)

   ![image](https://user-images.githubusercontent.com/6268044/158303693-34c18d3f-a8bd-4c72-88eb-97028ab4336b.png)

2. Open second terminal (`(all)---` is not redraw)

   ![image](https://user-images.githubusercontent.com/6268044/158303833-ebe42e72-eeb0-4f14-ae95-02e312498d6c.png)

3. Force redraw

   ![image](https://user-images.githubusercontent.com/6268044/158303905-8e7f1eee-d2f1-45c0-82de-b7ada654ba27.png)

4. Close second terminal (`(all)` is incorrectly redrawn)

   ![image](https://user-images.githubusercontent.com/6268044/158304015-9febc7fa-d622-466d-ac68-c03cebfac82e.png)

## NeoMutt version

```
NeoMutt 20211029-454-6adf99
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.16.0-4-amd64 (x86_64)
ncurses: ncurses 6.3.20211021 (compiled with 6.3.20211021)
libidn: 1.38 (compiled with 1.38)
GPGME: 1.16.0-unknown
GnuTLS: 3.7.3
libnotmuch: 5.6.0
storage: tokyocabinet

Configure options: --mandir=/usr/share/man --libexecdir=/usr/libexec --with-mailpath=/var/mail --gpgme --lua --notmuch --with-ui --gnutls --gss --idn --mixmaster --sasl --tokyocabinet --sqlite --autocrypt --debug-graphviz --asan --debug-window

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -I/usr/include/lua5.4 -DNCURSES_WIDECHAR -isystem /usr/include/mit-krb5 -fsanitize=address -O0 -fno-omit-frame-pointer -fno-common

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme +gss 
  +hcache -homespool +idn +inotify -locales_hack +lua +mixmaster +nls +notmuch 
  -openssl +pgp +regex +sasl +smime +sqlite +sun_attachment 

Devel options:
  asan graphviz window 

MAILPATH="/var/mail"
MIXMASTER="mixmaster"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

**Comment by gahr** [comment]
Might be the same root cause behind the difference in the messages shows in the status bar while opening a mailbox.

**master**
```
Fetching message headers
```

**23b03a69f**
```
Mailbox is unchanged                                                                                                                    
Reading /path/to/INBOX...                                                                                         
Selecting INBOX...                                                                                                                      
Evaluating cache...                                                                                                                     
Sorting mailbox...
```

**Comment by flatcap** [comment]
I've pushed a quick fix to [[devel/repaint]](https://github.com/neomutt/neomutt/tree/devel/repaint).
I'll open a PR when I've had a deeper look.

**Comment by dcpurton** [comment]
> I've pushed a quick fix to [[devel/repaint]](https://github.com/neomutt/neomutt/tree/devel/repaint). I'll open a PR when I've had a deeper look.

Thanks! This addresses what I'm seeing so far.

**Comment by flatcap** [comment]
@gahr 
> difference in the messages

The messages were being overwritten before anyone refreshed the screen.
Fixed in 0fac7e03c
